### DISCUSSION TITLE
Loading An Asset Twice Causes An Error

### DISCUSSION THREAD

**Discussion by N8python** [enable3d/enable3d#115] [Help / Questions] [answered]
I used the `this.scene.restart()` function to restart my scene, and this involves loading some assets a second time (or loading them from a cache). 
When I do this, I recieve the following error: ```index.js:1 Uncaught (in promise) TypeError: t.lastIndexOf is not a function```
This is the error traceback:
```
 at i.load (index.js:2)
    at loaders.js:137
    at new Promise (<anonymous>)
    at e.default.fbx (loaders.js:136)
```
This doesn't happen when I load the asset the first time - only when I do so the second time.

**Comment by yandeu** [comment, marked as answer]
You have to use `load.preload('key', 'url')` which will cache your assets.

Then, use `load.fbx('key')`.

---

An alternative solution would be `load.fbx('key', 'url')`, but it is not implemented yet.

**Reply by N8python** [reply]
I see - thank you very much! You are super available and you've really helped me learn the library.

**Reply by yandeu** [reply]
Well, I obviously should improve the [docs](https://enable3d.io/docs.html).
Feel free to add everything you learn to the docs 🥰

**Reply by N8python** [reply]
I will, once I'm more proficient with the library. Wouldn't want to add any misinformation to the docs.
